app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb

```diff
@@ -0,0 +1,146 @@
+module Authentication
+  module AuthnJwt
+    module VendorConfigurations
+      # Mock JWTConfiguration class to use it to develop other part in the jwt authenticator
+      class ConfigurationJWTGenericVendor < ConfigurationInterface
+
+        def initialize(
+          authenticator_input:,
+          logger: Rails.logger,
+          authentication_parameters_class: Authentication::AuthnJwt::AuthenticationParameters,
```

Comment:
This is actually something I don't love about Jonah's Command Class implementation (a small nit). Personally, I like the explicitness of initializing the class with it's arguments from the outside:
```
jwt_config = Authentication::AuthnJwt::VendorConfigurations::ConfigurationJWTGenericVendor.new(
  authenticator_input: authenticator_input,
  logger: Rails.logger,
  authentication_parameters_class: Authentication::AuthnJwt::AuthenticationParameters.new(
    authentication_input: <authn_input>, 
    jwt_token: <jwt_token>
  )
  ...
)
```
This forces you think about what is being passed.  It also prevents you from instantiating classes inside the command class (which is an anti-pattern).

This added benefit is testing becomes much easier.

Reply:
It can be discussed not sure if im understand